webhook/webhook.go

```diff
@@ -0,0 +1,97 @@
+package webhook
+
+import (
+  "crypto/hmac"
+  "crypto/sha256"
+  "encoding/hex"
+  "net/http"
+  "fmt"
+  "errors"
+  "encoding/json"
+  "bytes"
+  "time"
+
+  log "github.com/gophish/gophish/logger"
+)
+
+const (
+  DefaultTimeoutSeconds = 10
+  MinHttpStatusErrorCode = 400
+  SignatureHeader = "X-Gophish-Signature"
+)
+
+
+//TODO
+
+type Sender interface {
+  //TODO replace with EndPoint
+  Send(url string, secret string, data interface{}) error
+}
+
+type defaultSender struct {
+  client *http.Client
+}
+
+var senderInstance = &defaultSender {
+  client: &http.Client{
+    Timeout: time.Second * DefaultTimeoutSeconds,
+    CheckRedirect: func(req *http.Request, via []*http.Request) error {
+      return http.ErrUseLastResponse
+    },
+  },
+}
+
+type EndPoint struct {
+  Url string
```

Comment:
Minor nit - can we please make this `URL` to match `models/campaign.go`?

Reply:
renamed to URL